include/kos/stdlib.h

```diff
@@ -0,0 +1,30 @@
+/* KallistiOS ##version##
+
+   kos/stdlib.h
+   Copyright (C) 2023 Falco Girgis
+*/
+
+/* 
+   Add select POSIX extensions to stdlib.h which are not present within Newlib.
+*/ 
+
+#ifndef _STDLIB_H_
+#error "Do not include this file directly. Use <stdlib.h> instead."
+#endif /* !_STDLIB_H_ */
+
+#ifndef __KOS_STDLIB_H
+#define __KOS_STDLIB_H
+
+#if !defined(__STRICT_ANSI__) || (__STDC_VERSION__ >= 201112L) || (__cplusplus >= 201703L)
```

Comment:
I think this might need to be adjusted, since this will make the symbol appear in strict C11/C++17 mode, which I don't think is correct.

Reply:
Yeah, good point. So technically this is supposed to be available when `_POSIX_C_SOURCE >= 200112L`, but Newlib only has it defined up to 2009, and I didn't want to redefine it or anything since we don't actually support everything from 20011... So I was just going to enable the code whenever strict ANSI was not enabled.